File: api/pkg/model/rfc.go
package model

import (
	"gopkg.in/mgo.v2/bson"
)

// RFC (Request for Change) struct
type RFC struct {
	ID               bson.ObjectId `bson:"_id" json:"id"`
	OwnerID          string        `bson:"ownerID" json:"ownerID"`
	Status           Status        `bson:"status" json:"status"`
	Title            string        `bson:"title" json:"title"`
	Description      string        `bson:"description" json:"description"`
	ReleaseDate      float32       `bson:"releaseDate" json:"releaseDate"`
	ReleaseTime      string        `bson:"releaseTime" json:"releaseTime"`
	PullRequestLinks bool          `bson:"pullRequestLinks" json:"pullRequestLinks"`
	UserStoryLinks   string        `bson:"userStoryLinks" json:"userStoryLinks"`
	OutageRequired   bool          `bson:"outageRequired" json:"outageRequired"`
}

File: ui/src/model/RFC.ts

export interface RFC {
    id: string,
	ownerID: string,
	status: StatusEnum,
	title: string,
	description: string,
	releaseDate: string,
	releaseTime: string,
	pullRequestLinks: boolean,
	userStoryLinks: string,
	outageRequired: boolean
}

export interface Status {
	id: number,
	label: string,
	color: string,
	active: boolean
}

enum StatusEnum {
	QAApproval = 0,
	SREApproval,
	Approved,
	Rejected,
	Deployed
}

export const StatusText: Array<string> = ["Waiting QA Approval", "Waiting SRE Approval", "Approved", "Rejected", "Deployed"]

export const Statuses: Array<Status> = [
	{
		id: 0,
		label: "Waiting QA Approval",
		color: "warning",
		active: true
	},
	{
		id: 1,
		label: "Waiting SRE Approval",
		color: "secondary",
		active: false
	},
	{
		id: 2,
		label: "Approved",
		color: "success",
		active: false
	},
	{
		id: 3,
		label: "Rejected",
		color: "danger",
		active: true
	},
	{
		id: 4,
		label: "Deployed",
		color: "dark",
		active: false
	}
]